### PR TITLE
Update PR template with new branching strategy

### PR DESCRIPTION
**Pull request by dennisameling**
Please find attached the new PR template based on the new branching strategy: https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#step-5-work-on-your-pull-request

Also changed `m3.mautibox.com` back to `mautibox.com` because that's the URL to test M3 PRs now (m3. redirects to the root domain)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
+| -------------------------------------- | ---
+| Branch?                                | staging for features or enhancements / 3.0 for bug fixes <!-- see below -->
+| Bug fix?                               | yes/no
+| New feature?                           | yes/no
+| Deprecations?                          | yes/no
+| BC breaks?                             | yes/no
+| Automated tests included?              | yes/no
+| Related user documentation PR URL      | mautic/mautic-documentation#... <!-- required for new features -->
+| Related developer documentation PR URL | mautic/developer-documentation#... <!-- required for developer-facing changes -->
+| Issue(s) addressed                     | Fix #... <!-- prefix each issue number with "Fix #", no need to create an issue if none exist, explain below instead -->
```
Comment (RCheesley) [inline .github/PULL_REQUEST_TEMPLATE.md]: Just a small suggestion, as I think it reads a bit better!

```suggestion
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
-| Related developer documentation PR URL | 
-| Issues addressed (#s or URLs) | 
-| BC breaks? | 
-| Deprecations? | 
+<!--
+Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#step-5-work-on-your-pull-request):
+ - Always add tests and ensure they pass.
+ - Bug fixes must be submitted against the lowest maintained branch where they apply
+   (lowest branches are regularly merged to upper ones so they get the fixes too.)
+ - Features and deprecations must be submitted against branch staging.
```
Comment (RCheesley) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
 - Features and deprecations must be submitted against the staging branch.
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Just a couple of minor suggestions!

**Comment by dennisameling** [comment]
@RCheesley done, thanks! :)

**Review by RCheesley — APPROVED** [review]

**Review by npracht — APPROVED** [review]
Looks good to me :)

**Pull request merged**